src/System.Windows.Forms/src/System/Windows/Forms/CheckedListBox.cs

```diff
@@ -454,9 +454,10 @@ private void LbnSelChange()
         if (!_killnextselect && (index == _lastSelected || CheckOnClick))
         {
             CheckState currentValue = CheckedItems.GetCheckedState(index);
-            CheckState newValue = (currentValue != CheckState.Unchecked)
-                                  ? CheckState.Unchecked
-                                  : CheckState.Checked;
+            CheckState newValue =
+                currentValue != CheckState.Unchecked
+                ? currentValue != CheckState.Checked ? CheckState.Indeterminate : CheckState.Checked
+                : CheckState.Unchecked;
```

Comment:
Actually, on a second look, I think we should keep this part as it was before:
```suggestion
            CheckState newValue = (currentValue != CheckState.Unchecked)
                                  ? CheckState.Unchecked
                                  : CheckState.Checked;
```

`CheckedListBox` doesn't have a property like `ComboBox.ThreeState`, with which we could cycle through 3 values (checked-unchecked-indeterminate), so we should keep checked-unchecked cycle. Other controls with Indeterminate state support, but without ThreeState property (ToolStripMenuItem, ToolStripButton, or CheckBox with ThreeState = false and CheckState = CheckState.Indeterminate), also change Indeterminate to Unchecked and when cycling through don't set Indeterminate again.

Reply:
Thanks for your guidance.
From this bug, the tools can read both states of ToggleState, only indeterminate is missing, so we should support the indeterminate on ToggleState first. 
And then test the changes, if the test doesn't pass, we need to check the methods in CheckedListBox.cs again.